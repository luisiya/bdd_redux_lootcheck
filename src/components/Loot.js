import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchBitcoin } from "../actions/bitcoin";

export class Loot extends Component{
    componentDidMount() {
        this.props.fetchBitcoin();
    }
    getBitcoin = () => {
         const { bitcoin } = this.props;
         if (Object.keys(bitcoin).length === 0) return "";

         return this.props.balance / parseInt(bitcoin.bpi.USD.rate.replace(",", ""), 10);

    };
    render() {
        return(
            <h3 className="bitcoinBalance">Bitcoin Balance: {this.getBitcoin()}</h3>
        )
    }
}

export default connect(state => state, {fetchBitcoin} )(Loot);