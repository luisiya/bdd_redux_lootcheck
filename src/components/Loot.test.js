import React from "react";
import { mount, shallow } from "enzyme";
import { Loot } from "./Loot";

describe("Loot", () => {
    const mockFetchbitcoin = jest.fn();
    let props = { balance: 10, bitcoin: {}};
    let loot = shallow (<Loot {...props}/>);

    it("Loot renders correctly", () => {
       expect(loot).toMatchSnapshot();
    });

    describe("When Loot mounted", () => {
        beforeEach(() => {
            props.fetchBitcoin = mockFetchbitcoin;
            loot = mount(<Loot {...props}/>);
        });
        it("Dispatches fetchBitcoin method from props", () => {
            expect(mockFetchbitcoin).toHaveBeenCalled();
        });
    });

    describe('When thre are valid bitcoin props', function () {
        beforeEach(() => {
            props = { balance: 10, bitcoin: {bpi: {USD: {rate: "1,000"}}}};
            loot = shallow (<Loot {...props}/>);
        });

            it('should displays correct bitcoin value', function () {
                expect(loot.find(".bitcoinBalance").text()).toEqual("Bitcoin Balance: 0.01");
            });

    });
});