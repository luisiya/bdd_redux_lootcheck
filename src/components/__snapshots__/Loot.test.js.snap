// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot Loot renders correctly 1`] = `ShallowWrapper {}`;
